core/src/main/java/com/linecorp/armeria/internal/server/HttpDocServicePlugin.java

```diff
@@ -0,0 +1,143 @@
+ */
+public final class HttpDocServicePlugin extends AbstractDocServicePlugin {
+
+    @VisibleForTesting
+    static final TypeSignature HTTP_RESPONSE = TypeSignature.ofBase(HttpResponse.class.getSimpleName());
+
+    static final Set<HttpMethod> IGNORED_HTTP_METHODS = ImmutableSet.of(HttpMethod.CONNECT);
+
+    static final Map<HttpMethod, String> METHOD_NAMES =
+            HttpMethod.knownMethods()
+                      .stream()
+                      .filter(method -> !IGNORED_HTTP_METHODS.contains(method))
+                      .collect(toImmutableMap(method -> method,
+                                              method -> "do" + capitalize(method.name().toLowerCase())));
+
+    private static String capitalize(String str) {
+        return str.substring(0, 1).toUpperCase() + str.substring(1);
+    }
+
+    @Override
+    public String name() {
+        return "http";
+    }
+
+    @Override
+    public Set<Class<? extends Service<?, ?>>> supportedServiceTypes() {
+        return ImmutableSet.of(AbstractHttpService.class);
+    }
+
+    @Override
+    public ServiceSpecification generateSpecification(Set<ServiceConfig> serviceConfigs,
+                                                      DocServiceFilter filter) {
+        requireNonNull(serviceConfigs, "serviceConfigs");
+        requireNonNull(filter, "filter");
+
+        final Map<Class<?>, Set<MethodInfo>> methodInfos = new HashMap<>();
+        for (ServiceConfig sc : serviceConfigs) {
+            final AbstractHttpService service = sc.service().as(AbstractHttpService.class);
+            if (service == null) {
+                continue;
+            }
+            final String className = service.getClass().getName();
+            for (HttpMethod method : sc.route().methods()) {
+                if (IGNORED_HTTP_METHODS.contains(method)) {
+                    continue;
+                }
+                final String methodName = METHOD_NAMES.get(method);
+                if (!filter.test(name(), className, methodName)) {
+                    continue;
+                }
+                addMethodInfo(methodInfos, sc.virtualHost().hostnamePattern(), service, sc.route(),
+                              method);
+            }
+        }
+        return generate(ImmutableMap.of(), methodInfos);
+    }
+
+    @Override
+    public String toString() {
+        return HttpDocServicePlugin.class.getSimpleName();
+    }
+
+    private static void addMethodInfo(Map<Class<?>, Set<MethodInfo>> methodInfos, String hostnamePattern,
+                                      AbstractHttpService service, Route route, HttpMethod httpMethod) {
+        final EndpointInfo endpoint = endpointInfo(route, hostnamePattern);
+        final Class<?> clazz = service.getClass();
+        final String methodName = METHOD_NAMES.get(httpMethod);
+        final MethodInfo methodInfo = new MethodInfo(
+                methodName, HTTP_RESPONSE, ImmutableList.of(), ImmutableList.of(), // Ignore exceptions.
```

Comment:
How about defining a static final field list that has `ServiceRequestContext` and `HttpRequest` and passes it here as a parameter?